### DISCUSSION TITLE
picoprobe setup?

### DISCUSSION THREAD

**Discussion by madskjeldgaard** [mfussenegger/nvim-dap#454] [Q&A]
Hello, I'm trying to [emulate this setup](https://www.digikey.be/en/maker/projects/raspberry-pi-pico-and-rp2040-cc-part-2-debugging-with-vs-code/470abc7efb07432b82c95f6f67f184c0) for setting up a debug adapter that will make it possible to debug Raspberry Pi Pico code. However, I am stuck and cannot figure out what to do (newbie to dap, sorry). Here's how far I got:

```lua
-- For Picoprobe
dap.adapters.cortexdebug = {
	type = 'executable',
	command = '/usr/bin/arm-none-eabi-gdb',
	name = "arm-none-eabi-gdb"
}

dap.configurations.cpp = {
	{
		name = "DebugPico",
		type = "cortexdebug",
		request = "launch",
		program = function()
			return vim.fn.input('Path to executable: ', vim.fn.getcwd() .. '/', 'file')
		end,
		cwd = '${workspaceFolder}',
		stopOnEntry = false,
		args = {"--eval-command=\"target extended-remote :3333\""},
		-- FIXME:
		-- executable = "${command:cmake.launchTargetPath}",
		-- FIXME:
		servertype = "openocd",
		gdbPath =  "arm-none-eabi-gdb",
		device = "RP2040",
		configFiles = { "interface/picoprobe.cfg", "target/rp2040.cfg" },
		svdFile = "/usr/share/pico-sdk/src/rp2040/hardware_regs/rp2040.svd",
		runToMain = true,
		-- Work around for stopping at main on restart
		postRestartCommands = { "break main", "continue" },
		-- FIXME:
		-- searchDir = ["<PATH TO TCL>"],
		runInTerminal = true,
	}
}
```
when running `continue`, dap says that the adapter did not respond. Any pointers? it's a rather complicated place to start, I guess but I really need this setup in my projects at the moment :)

**Comment by mfussenegger** [comment]
Maybe the discussion in  https://github.com/mfussenegger/nvim-dap/discussions/434 helps?
